src/nodes.coffee

```diff
@@ -2089,11 +2105,16 @@ exports.Code = class Code extends Base
             ref = param.asReference o
           else
             if param.value? and not param.assignedInBody
-              ref = new Assign new Value(param.name), param.value
+              ref = new Assign new Value(param.name), param.value, null, param: yes
             else
               ref = param
-          # Add this parameter’s reference to the function scope
-          o.scope.parameter fragmentsToText (if param.value? then param else ref).compileToFragments o
+          # Add this parameter’s reference(s) to the function scope.
+          if param.name instanceof Arr or param.name instanceof Obj
+            # This parameter is destructured.
+            param.name.eachName (prop) ->
+              o.scope.parameter fragmentsToText prop.compileToFragments o
```

Comment:
This could use the simpler logic from `Assign` (e.g. `o.scope.parameter prop.value`).